src/cobalt.rs

```diff
@@ -174,25 +170,34 @@ fn generate_pages(posts: Vec<Document>, pages: Vec<Document>, context: &Context)
     trace!("Generating other documents");
     for mut doc in pages {
         trace!("Generating {} / {:?}", doc.url_path, doc.file_path.to_str());
-        let mut pagination_cfg = pagination_config::PaginationCfg::new(&doc.front.pagination);
-        if pagination_cfg.is_pagination_enable() {
+        let pagination_enabled = {
+            doc.front
+                .pagination
+                .as_ref()
+                .map_or(false, |pagination_config| {
+                    pagination_config.is_pagination_enable()
```

Comment:
If we are accessing a non-config version of the paginator, then we could just check whether `opt.is_some()` or even `if let Some(pagination) = ...`.

Reply:
I think I'll end up with the compilation error I've asked about on the Rust forum. I'll try though.